debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java

```diff
@@ -301,19 +320,44 @@ private ValueConverter createArrayConverter(Column column, Field fieldDefn) {
         return data -> convertArray(column, fieldDefn, elementConverter, data);
     }
 
-    @Override
-    protected Object convertDecimal(Column column, Field fieldDefn, Object data) {
-        BigDecimal newDecimal = (BigDecimal) super.convertDecimal(column, fieldDefn, data);
-        if (newDecimal == null) {
-            return newDecimal;
+    protected Object convertDecimal(Column column, Field fieldDefn, Object data, DecimalMode mode) {
+        DebeziumDecimal value;
+        BigDecimal newDecimal;
+
+        if (data == null || !(data instanceof DebeziumDecimal)) {
+            newDecimal = (BigDecimal)super.convertDecimal(column, fieldDefn, data);
+            if (newDecimal == null) {
+                return newDecimal;
+            }
+            value = new DebeziumDecimal(newDecimal);
+        }
+        else {
+            value = (DebeziumDecimal)data;
+        }
+
+        if (!value.getDecimalValue().isPresent()) {
+            if (mode != DecimalMode.STRING) {
+                logger.warn("Got a special value (NaN/Infinity) for Decimal type but current mode is not handle it. "
+                        + "If you need to support it then set decimal handling mode to 'string'.");
+                return isVariableScaleDecimal(column) ? VariableScaleDecimal.fromLogical(fieldDefn.schema(), DebeziumDecimal.ZERO) : BigDecimal.ZERO;
```

Comment:
I don't think returning zero is a good idea. Let's rather raise an exception, requiring people to adjust their config. That seems better than losing values.